heartbeat/monitors/monitor.go

```diff
@@ -76,16 +76,9 @@ func checkMonitorConfig(config *common.Config, registrar *plugin.PluginsReg) err
 	return err
 }
 
-// uniqueMonitorIDs is used to keep track of explicitly configured monitor IDs and ensure no duplication within a
+// globalDedup is used to keep track of explicitly configured monitor IDs and ensure no duplication within a
 // given heartbeat instance.
-var uniqueMonitorIDs sync.Map
-
-// ErrDuplicateMonitorID is returned when a monitor attempts to start using an ID already in use by another monitor.
-type ErrDuplicateMonitorID struct{ ID string }
-
-func (e ErrDuplicateMonitorID) Error() string {
-	return fmt.Sprintf("monitor ID %s is configured for multiple monitors! IDs must be unique values.", e.ID)
-}
+var globalDedup = newDedup()
```

Comment:
Could this be part of the `RunnerFactory` to avoid needing a global object?

Reply:
Great idea, done!